### PR TITLE
README Update

### PR DESCRIPTION
**Pull request by Darshini-Parikh93**
- Updated top level README for protocol samples folder structure change.
- Added steps to run component sample services.
- Added runtime properties used by each service in component-samples
and their default values.
- Added README for demo folder with necessary docker commands
- Updated README for each component in demo folder.

Signed-off-by: Darshini Parikh <darshini.k.parikh@intel.com>

### CODE REVIEW

File: demo/device/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+```
+$ cd <fido-iot-src>/service/component-samples/http-device-sample/
+$ mvn clean install
+```
+
+This will copy the required executables and libraries into <fido-iot-src>/demo/device/.
+
+# Starting the Device service
+
+Refer the [Device Component Samples](../../README.md/#running-the-fido-iot-http-java-device-sample) section to start the device.
```
Comment (trbehera) [inline demo/device/README.md:13]: The demo component would also be released as a stand-alone binary. We should be including all necessary information without having to refer to other source README files.

A link to demo README from other source README should be okay, though.

File: service/component-samples/http-device-sample/pom.xml [hunk 2]
```diff
@@ -193,11 +193,17 @@
         <configuration>
           <filesets>
             <fileset>
-              <directory>${project-demo-directory}/${project.artifactId}</directory>
+              <directory>${project-demo-directory}/${project.artifactId}/lib</directory>
```
Comment (trbehera) [inline service/component-samples/http-device-sample/pom.xml:196]: Please raise a different PR for this fix as this is not related to README.

**Review by trbehera — CHANGES_REQUESTED** [review]

File: demo/device/README.md (hunk 1 above)
Comment (Darshini-Parikh93) [reply demo/device/README.md]: Done

File: service/component-samples/http-device-sample/pom.xml (hunk 2 above)
Comment (Darshini-Parikh93) [reply service/component-samples/http-device-sample/pom.xml]: Removed from this PR.

**Review by trbehera — APPROVED** [review]

**Review by prateekcr — APPROVED** [review]
